addons/stock/models/stock_move.py

```diff
@@ -162,8 +162,9 @@ def _default_group_id(self):
 
     @api.depends('picking_id.is_locked')
     def _compute_is_locked(self):
+        # XDU This could be improved as it can take up to 2s when 300 moves are computed.
```

Comment:
> 2s when 300 moves are computed

how is it possible? can you provide a flamegraph or something?

Reply:
@sle-odoo  The thing is that when editing one line, the onchange is triggered on the field "move_line_ids" of the stock_picking so all the lines are recomputed even if only one has changed.

You can see the profiling here : https://drive.google.com/file/d/1bRi-aSoT8w-BUxrcoBvRUtN3dLGMQGJi/view?usp=sharing

Reply:
Then maybe the issue is at a lower level: does that make sense to change the value of a non-stored computed field? Because that globally what you do in this PR.

Reply:
@nim-odoo it is a compute because the locked concept is re-used in mrp [here](https://github.com/odoo/odoo/blob/12.0/addons/mrp/models/stock_move.py#L104)

Reply:
Or something like:
```python
picking_ids = [r[0] for r in self.read(['picking_id'])]
is_locked = [r[0] for r in self.env['stock.picking'].browse([picking_ids]).read(['is_locked'])]
if all([r == False for r in is_locked]):
    return
```

You could also retrieve the value thanks to a SQL request, but not sure it is a good idea in a computed field.

Reply:
From what I checked, the main problem is that all the 300 moves are triggered via the onchange even if only one line have been modified and that the picking "is locked" field has not changed... So we could improve the computing but the real solution would be to prevent this computing, no ?

Reply:
We already had this conversation, and long story short: no, we shouldn't. There is the same issue in the inventory view.

Reply:
Is not the same issue that I have reported here:
 - https://github.com/odoo/odoo/issues/30578